src/python/Utils/Throttled.py

```diff
@@ -16,41 +36,131 @@
 
 class _ThrottleCounter(object):
     """
-    _ThrottleCounter class define throttle parameter and
+    _ThrottleCounter class defines throttle parameter and
     enter/exit methods to work with `with` context.
     """
 
     def __init__(self, throttle, user, debug=False):
         self.throttle = throttle
         self.user = user
-        self.debug = debug
+        if debug:
+            self.throttle.logger.setLevel(logging.DEBUG)
 
     def __enter__(self):
         "Define enter method for `with` context"
         ctr = self.throttle._incUser(self.user)
-        if self.debug:
-            msg = "Entering throttled function with counter %d for user %s" \
-                    % (ctr, self.user)
-            self.throttle.logger.debug(msg)
+	msg = "Entering throttled function with counter %d for user %s" \
+		% (ctr, self.user)
+	self.throttle.logger.debug(msg)
         if ctr >= self.throttle.getLimit():
             self.throttle._decUser(self.user)
             msg = "The current number of active operations for this resource"
             msg += " exceeds the limit of %d for user %s" \
                    % (self.throttle.getLimit(), self.user)
-            raise cherrypy.HTTPError("500, Internal Server Error", msg)
+            raise cherrypy.HTTPError("429 Too Many Requests", msg)
 
     def __exit__(self, exc_type, exc_value, exc_traceback):
         "Define exit method for `with` context"
         ctr = self.throttle._decUser(self.user)
-        if self.debug:
-            msg = "Exiting throttled function with counter %d for user %s" \
-                  % (ctr, self.user)
-            self.throttle.logger.debug(msg)
+	msg = "Exiting throttled function with counter %d for user %s" \
+	      % (ctr, self.user)
+	self.throttle.logger.debug(msg)
 
+class _ThrottleTimeCounter(object):
+    """
+    _ThrottleTimeCounter class defines time range throttled mechanism
+    """
+
+    def __init__(self, throttle, user, trange):
+        self.throttle = throttle
+        self.user = user
+        self.trange = trange
+
+    def __enter__(self):
+        "Define enter method for `with` context"
+        ctr = self.throttle._incUser(self.user, self.trange)
+        if ctr > self.throttle.getLimit():
+            self.throttle._decUser(self.user)
```

Comment:
same comment as above.

Reply:
not fixed since __exit__ does not decrement counter. Here the decrement call is correct